src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -31,63 +70,134 @@ public ConsoleLoggerProcessor(IConsole console, IConsole errorConsole)
                 Name = "Console logger queue processing thread"
             };
             _outputThread.Start();
+            s_meter.CreateObservableGauge<long>("queue-size", GetQueueSize);
         }
 
         public virtual void EnqueueMessage(LogMessageEntry message)
         {
-            if (!_messageQueue.IsAddingCompleted)
-            {
-                try
-                {
-                    _messageQueue.Add(message);
-                    return;
-                }
-                catch (InvalidOperationException) { }
-            }
-
-            // Adding is completed so just log the message
-            try
+            // cannot enqueue when adding is completed
+            if (!Enqueue(message))
             {
                 WriteMessage(message);
             }
-            catch (Exception) { }
         }
 
         // for testing
         internal void WriteMessage(LogMessageEntry entry)
         {
-            IConsole console = entry.LogAsError ? ErrorConsole : Console;
-            console.Write(entry.Message);
+            try
+            {
+                var messagesDropped = Interlocked.Exchange(ref _messagesDropped, 0);
+                if (messagesDropped != 0)
+                {
+                    System.Console.Error.WriteLine($"{messagesDropped} message(s) dropped because of queue size limit. Increase the queue size or decrease logging verbosity to avoid this.{Environment.NewLine}");
```

Comment:
I think we should write this message to the IConsole and not to Console.Error. I also suggest that we write this message in exactly the position in the text where the other log messages have been omitted which would mean the warning message itself gets enqueued:

```C#
                 //in the Enqueue function
                if (_messageQueue.Count < MaxQueueLength)
                {
                    bool startedEmpty = _messageQueue.Count == 0;
                    if(_messagesDropped > 0)
                    {
                        _messageQueue.Enqueue(warningText);
                        _messagesDropped=0;
                    }
                    // if we just logged the dropped message warning this could push the queue size to 
                    // MaxLength+1, that shouldn't be a problem. It won't grow any further until it is less than
                    // MaxLength once again.
                    _messageQueue.Enqueue(item);
                    // if the queue started empty it could be at 1 or 2 now
                    if (startedEmpty)
                    {
                        // pulse for wait in Dequeue
                        Monitor.PulseAll(_messageQueue);
                    }

```